tests/test_tng.py

```diff
@@ -121,88 +123,109 @@ def test_tng_example_getitem_bool_TypeError(cls, TNG_EXAMPLE):
             for ts in tng[slx]:
                 ts.step
 
-@pytest.mark.skip(reason="FAILING")
+
 def test_tng_example_natoms(TNG_EXAMPLE_DATA, TNG_EXAMPLE):
     with pytng.TNGFileIterator(TNG_EXAMPLE) as tng:
         assert TNG_EXAMPLE_DATA.natoms == tng.n_atoms
 
+
+def test_README_example(TNG_EXAMPLE):
+    with pytng.TNGFileIterator(TNG_EXAMPLE, 'r') as tng:
+        positions = np.ndarray((tng.n_atoms, 3), dtype=np.float32)
+        for ts in tng:
+            time = ts.get_time()
+            positions = ts.get_positions(positions)
+
+
+def test_DOCS_example(TNG_EXAMPLE):
+    with pytng.TNGFileIterator(TNG_EXAMPLE, 'r') as tng:
+        positions = tng.make_ndarray_for_block_from_name("TNG_TRAJ_POSITIONS")
+        for ts in tng[0:len(tng):tng.block_strides["TNG_TRAJ_POSITIONS"]]:
+            positions = ts.get_positions(positions)
+
+
 def test_tng_example_tng_example_first_positions(
     TNG_EXAMPLE_DATA, TNG_EXAMPLE
 ):
     with pytng.TNGFileIterator(TNG_EXAMPLE) as tng:
-        print(tng.block_strides)
-        first_frame = tng[0].pos
-        print(first_frame)
-        print(TNG_EXAMPLE_DATA.first_frame)
-        assert np.array_equal(TNG_EXAMPLE_DATA.first_frame, first_frame)
+        pos = np.zeros((15, 3), dtype=np.float32)
+        pos = tng[0].get_positions(pos)
+        assert np.array_equal(TNG_EXAMPLE_DATA.first_frame, pos)
+
 
 def test_tng_example_tng_example_last_positions(TNG_EXAMPLE_DATA, TNG_EXAMPLE):
     with pytng.TNGFileIterator(TNG_EXAMPLE) as tng:
-        
-        last_frame = tng[len(tng)-1].pos
-        assert np.array_equal(TNG_EXAMPLE_DATA.last_frame, last_frame)
+        pos = np.zeros((15, 3), dtype=np.float32)
+        pos = tng[9].get_positions(pos)
```

Comment:
is it worth parametrizing the indices tested to include a `-1` for reverse index support to access the last element?

Reply:
Negative indices not currently supported. I should add them in.

Is there a canonical way to do that with an Iterator or should I implement my own version?

Reply:
Fixed for proper wraparound on the iterator.